calicoctl/calico_ctl/node.py

```diff
@@ -847,8 +848,13 @@ def handle_sigterm(sig, frame):
     except KeyboardInterrupt:
         # Mainline. Someone pressed Ctrl-C.
         print "Stopping Calico node..."
+        clean_exit = True
     finally:
         # Could either be this process is being killed, or output generator
         # raises an exception.
         if stop_container_on_exit:
             docker_client.stop(container)
+            # If the container is stopped due to unknown circumstances, exit
+            # with failed status code.
+            if not clean_exit:
+                sys.exit(1)
```

Comment:
The docker-py library has a method called `wait` that is identical to `attach`, but instead of returning stdout/stderr on exit, it returns the return code of the container. I did a quick test and running `wait` on a stopped container will immediately return the ret code.

I think a nicer solution would be to just `sys.exit(docker_client.wait(container))`